public/pages/PolicyDetails/containers/PolicyDetails/PolicyDetails.tsx

```diff
@@ -0,0 +1,243 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+import React, { Component } from "react";
+import {
+  EuiSpacer,
+  EuiTitle,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiButton,
+  EuiOverlayMask,
+  EuiButtonEmpty,
+  EuiModalFooter,
+  EuiModal,
+  EuiModalHeader,
+  EuiModalHeaderTitle,
+  EuiModalBody,
+  EuiCodeBlock,
+} from "@elastic/eui";
+import { RouteComponentProps } from "react-router-dom";
+import queryString from "query-string";
+import { PolicyService } from "../../../../services";
+import { BREADCRUMBS, ROUTES } from "../../../../utils/constants";
+import { getErrorMessage } from "../../../../utils/helpers";
+import PolicySettings from "../../components/PolicySettings/PolicySettings";
+import { ISMTemplate } from "../../../../../models/interfaces";
+import { CoreServicesContext } from "../../../../components/core_services";
+import DeleteModal from "../../components/DeleteModal/DeleteModal";
+import States from "../../../VisualCreatePolicy/components/States";
+
+interface PolicyDetailsProps extends RouteComponentProps {
+  policyService: PolicyService;
+}
+
+interface PolicyDetailsState {
+  policyId: string;
+  channelId: string;
+  primaryTerm: number;
+  lastUpdated: string;
+  description: string;
+  sequenceNumber: number;
+  schemaVersion: number;
+  ismTemplates: ISMTemplate[];
+
+  isJSONModalOpen: boolean;
+  policyJSON: any;
+  isDeleteModalVisible: boolean;
+}
+
+export default class PolicyDetails extends Component<PolicyDetailsProps,
+  PolicyDetailsState> {
+    static contextType = CoreServicesContext;
+    constructor(props: PolicyDetailsProps) {
+      super(props);
+
+      this.state = {
+        policyId: "",
+        channelId: "",
+        primaryTerm: -1,
+        lastUpdated: "",
+        description: "",
+        sequenceNumber: -1,
+        schemaVersion: -1,
+        ismTemplates: [],
+
+        isJSONModalOpen: false,
+        policyJSON: {policy:{states:[]}},
+        isDeleteModalVisible: false,
+      };
+    }
+
+    componentDidMount = async (): Promise<void> => {
+      this.context.chrome.setBreadcrumbs([BREADCRUMBS.INDEX_MANAGEMENT, BREADCRUMBS.INDEX_POLICIES]);
+      const { id } = queryString.parse(this.props.location.search);
+      if (typeof id === "string") {
+        this.context.chrome.setBreadcrumbs([BREADCRUMBS.INDEX_MANAGEMENT, BREADCRUMBS.INDEX_POLICIES, { text: id }]);
+        this.props.history.push(`${ROUTES.POLICY_DETAILS}?id=${id}`);
```

Comment:
Confused on this.. so based on this code we are already on ROUTES.POLICY_DETAILS because this component got mounted. And it has an id in the query string already because we are in a check that confirmed it. And then we are pushing to this same page again? Do we need this? I might be missing something.

Reply:
Modeled after the Rollup Details page, which has a similar line. It doesn't appear to do anything though, removed.